components/director/internal/formationmapping/handler.go

```diff
@@ -152,7 +152,7 @@ func (h *Handler) UpdateFormationAssignmentStatus(w http.ResponseWriter, r *http
 		return
 	}
 
-	if reqBody.State != model.CreateErrorAssignmentState && reqBody.State != model.ReadyAssignmentState && reqBody.State != model.ConfigPendingAssignmentState {
+	if len(reqBody.State) > 0 && reqBody.State != model.CreateErrorAssignmentState && reqBody.State != model.ReadyAssignmentState && reqBody.State != model.ConfigPendingAssignmentState {
```

Comment:
The `validation.In` rule (and all rules in the library) are implemented in a way to treat empty values as valid. There is a specific rule `validation.Required` that you need to add for fields that should be non-empty. This is documented in the godoc of each rule (for the `validation.In` as well).

```
// An empty value is considered valid. Use the Required rule to make sure a value is not empty.
```

I don't think we need the `validation.Empty` rule, even more, I think that this makes everything misleading, since if someone wants to make it required he will go and remove the `validation.Empty` option. Which won't do anything and it is just a trap.

I agree that it is hidden, however, It is an opinionated way that the library owner got when modeling the rules. We cannot overcome it and we just need to get used to it. The proposed idea makes it more hidden and misleading from my PoV.

Reply:
Yep, it makes sense if that's the initial approach of the library to model the rules. I miss that.